src/generator/models.ts

```diff
@@ -423,3 +287,229 @@ function generateParamGroupStruct(lang: Language, params: Parameter[]): StructDe
   }
   return st;
 }
+
+function generateUnmarshallerForResponseEnvelope(structDef: StructDef) {
+  // if the response envelope contains a discriminated type we need an unmarshaller
+  let found = false;
+  for (const prop of values(structDef.Properties)) {
+    if (prop.isDiscriminator) {
+      found = true;
+      break;
+    }
+  }
+  if (!found) {
+    return;
+  }
+  const receiver = structDef.Language.name[0].toLowerCase();
+  let unmarshaller = `func (${receiver} *${structDef.Language.name}) UnmarshalJSON(data []byte) error {\n`;
+  // add a custom unmarshaller to the response envelope
+  // find the discriminated type field
+  let field = 'FIND';
```

Comment:
Is FIND the default name for the discriminated type field?

Reply:
Or will this never happen and is just a filler value? If so, how do we know that this never happens?

Reply:
It's a filler value and should never happen.  In the event that it does, the resultant code won't compile.

Reply:
Although we could rework this part to throw an error in that case instead (might be better).

Reply:
Hmmm maybe throwing would be better, then we know immediately that there's a problem with a swagger?

Reply:
I will change it to throw (it's likely less a swagger issue, more likely a bug in the generator).

Reply:
Fixed